modules/RTC/MungeLocalSdp.js

```diff
@@ -0,0 +1,295 @@
+                            : this.pc.simulcast.ssrcCache;
+                    if (!requiredSSRCs.length) {
+                        logger.error(
+                            "No SSRCs stored for: "
+                                + videoTrack + " in " + this.pc);
+                        return;
+                    }
+                    if (!transformer.getSSRCCount()) {
+                        logger.error(
+                            "No video SSRCs found "
+                                + "(should be at least the recv-only one");
+                        return;
+                    }
+
+                    modified = true;
+
+                    // We need to fake sendrecv
+                    transformer.mediaDirection = "sendrecv";
+
+                    // Check if the recvonly has MSID
+                    const primarySSRC = requiredSSRCs[0];
+                    // FIXME the cname could come from the stream, but may
+                    // turn out to be too complex. It is fine to come up
+                    // with any value, as long as we only care about
+                    // the actual SSRC values when deciding whether or not
+                    // an update should be sent
+                    const primaryCname = "injected-" + primarySSRC;
+
+                    requiredSSRCs.forEach((ssrcNum) => {
+                        // Remove old attributes
+                        transformer.removeSSRC(ssrcNum);
+                        // Inject
+                        logger.debug(
+                            "Injecting video SSRC: " + ssrcNum
+                                + " for " + videoTrack);
+                        transformer.addSSRCAttribute({
+                            id: ssrcNum,
+                            attribute: 'cname',
+                            value: primaryCname
+                        });
+                        transformer.addSSRCAttribute({
+                            id: ssrcNum,
+                            attribute: 'msid',
+                            value: videoTrack.storedMSID
+                        });
+                    });
+                    if (requiredSSRCs.length > 1) {
+                        const group = {
+                            ssrcs: requiredSSRCs.join(" "),
+                            semantics: "SIM"
+                        };
+                        if (!transformer.findGroup(
+                                group.semantics, group.ssrcs)) {
+                            // Inject the group
+                            logger.debug(
+                                "Injecting SIM group for " + videoTrack, group);
+                            transformer.addSSRCGroup(group);
+                        }
+                    }
+                    // Insert RTX
+                    // FIXME in P2P RTX is used by Chrome regardless of this
+                    // option status
+                    if (!this.pc.options.disableRtx) {
```

Comment:
this has a lot of overlap with the stuff in RtxModifier (and other things in the setLocalDescription transform chain in TPC), i think we should find a way to re-use the logic so if we find bugs or make changes we don't have to do it in two places